src/main/scala/LaunchDetectorLambda.scala

```diff
@@ -1,35 +1,56 @@
-import java.io.{InputStream, OutputStream}
-
-import org.apache.commons.codec.digest.DigestUtils
-import com.amazonaws.services.kinesis.clientlibrary.types.UserRecord
-import com.amazonaws.services.kinesis.model.Record
-import com.amazonaws.services.lambda.runtime.events.KinesisEvent
-import com.amazonaws.services.lambda.runtime
-import com.gu.contentatom.thrift.Atom
+import com.amazonaws.services.lambda.runtime.events.SNSEvent
+import com.amazonaws.services.lambda.runtime.events.SNSEvent.SNSRecord
 import com.gu.crier.model.event.v1._
 import com.amazonaws.services.lambda.runtime.{Context, RequestHandler}
 
 import scala.collection.JavaConverters._
+import scala.concurrent.{Await, Future}
+import scala.concurrent.duration._
+import scala.util.{Failure, Success}
+import scala.concurrent.ExecutionContext.Implicits.global
+import org.apache.logging.log4j.scala.Logging
 
-class LaunchDetectorLambda extends RequestHandler[KinesisEvent, Unit] {
-  override def handleRequest(event:KinesisEvent, context: Context): Unit = {
-    val rawRecords: List[Record] = event.getRecords.asScala.map(_.getKinesis).toList
-    val userRecords = UserRecord.deaggregate(rawRecords.asJava)
+class LaunchDetectorLambda extends RequestHandler[SNSEvent, Unit] with Logging {
+  override def handleRequest(incomingEvent:SNSEvent, context: Context): Unit = {
+    val rawRecords: List[SNSRecord] = incomingEvent.getRecords.asScala.toList
 
-    println(s"Processing ${userRecords.size} records ...")
-    CrierEventProcessor.process(userRecords.asScala) { event=>
+    logger.info(s"Processing ${rawRecords.size} records ...")
+    val processingResults = SNSEventProcessor.process(rawRecords) { event=>
       event.itemType match {
         case ItemType.Atom=>
-          event.payload.exists({
-            case EventPayload.Atom(atom)=>
-              AtomEventProcessor.process(atom, event.eventType)
-            case _=>
-              false
-          })
+          PlutoSender.sendToVidispine(event)
         case _=>
           println("This event is for something other than an atom, not going to do anything.")
-          false
+          Future(Success(false))
       }
     }
+
+    val results = Await.result(processingResults, (context.getRemainingTimeInMillis-100).millis)
+    //Success(true) means that we processed it
```

Comment:
It might be nice to use `Either` for results throughout the code. At the moment it's a mixture of `Either`, `Try` and booleans.